From ea18772952b7a30f35db1c95194ec3ade6465f89 Mon Sep 17 00:00:00 2001
From: Samuel Verschelde <stormi-xcp@ylix.fr>
Date: Mon, 22 Sep 2025 18:32:10 +0200
Subject: [PATCH] Adapt /etc/rsyslog.d/xenserver.conf header for XCP-ng

* Mention XCP-ng in addition to XenServer. I didn't completely
  replace XenServer with XCP-ng. After all, the file is still named
  xenserver.conf, so that would be awkward.
* Do NOT invite users to edit /etc/rsyslog.d/remote.conf, which is
  managed by XAPI.
* Discourage any custom modifications, but in the case where it's
  unavoidable, redirect them to /etc/rsyslog.d/custom.conf,
  after validation by technical support.

Signed-off-by: Samuel Verschelde <stormi-xcp@ylix.fr>
---
 src/common/etc/rsyslog.d/xenserver.conf | 13 ++++++++++---
 1 file changed, 10 insertions(+), 3 deletions(-)

diff --git a/src/common/etc/rsyslog.d/xenserver.conf b/src/common/etc/rsyslog.d/xenserver.conf
index d59c633..ff08c48 100644
--- a/src/common/etc/rsyslog.d/xenserver.conf
+++ b/src/common/etc/rsyslog.d/xenserver.conf
@@ -1,7 +1,14 @@
 # -- DO NOT EDIT THIS FILE --
-# xenserver.conf contains the XenServer specific logging configurations
-# Any remote forwarding should be configured in another file in /etc/rsyslog.d/
-# e.g. /etc/rsyslog.d/remote.conf
+# xenserver.conf contains the XenServer/XCP-ng specific logging configurations
+#
+# Remote forwarding must be configured via Xen Orchestra, which will ask XAPI to
+# generate file /etc/rsyslog.d/remote.conf (do not edit that file either).
+#
+# Any other rsyslog configuration change must be avoided.
+# However, if there are no other options, they must be evaluated with technical support
+# representatives, and written to a file named /etc/rsyslog.d/custom.conf.
+# Note that there are no guarantees regarding the order in which rules from different
+# files are applied.
 
 # Suppress duplicate messages and report "Last line repeated n times"
 $RepeatedMsgReduction on
